gh-139482: speed up `os.environ.clear` by 2x (#139483)
python/cpython: Misc/NEWS.d/next/Library/2025-10-01-17-45-27.gh-issue-139482.yDMeEa.rst

> Make os.environ.clear() more efficient (linear complexity instead of quadratic)
> 

Reviewer: ```suggestion
Ensure that :data:`os.environ.clear() <os.environ>`
has linear complexity instead of quadratic complexity.
```
